pkg/ddc/base/operation.go

```diff
@@ -0,0 +1,197 @@
+	case common.PhaseComplete:
+		return t.reconcileComplete(ctx, object, opStatus, operation)
+	case common.PhaseFailed:
+		return t.reconcileFailed(ctx, object, operation)
+	default:
+		ctx.Log.Error(fmt.Errorf("unknown phase"), "won't reconcile it", "phase", opStatus.Phase)
+		return utils.NoRequeue()
+	}
+}
+
+func (t *TemplateEngine) reconcileNone(ctx cruntime.ReconcileRequestContext, object client.Object,
+	opStatus *datav1alpha1.OperationStatus, operation dataoperation.OperationInterface) (ctrl.Result, error) {
+	log := ctx.Log.WithName("reconcileNone")
+
+	// 0. check the object spec valid or not
+	conditions, err := operation.Validate(ctx, object)
+	if err != nil {
+		log.Error(err, "validate failed")
+		ctx.Recorder.Event(object, v1.EventTypeWarning, common.DataOperationNotValid, err.Error())
+
+		opStatus.Conditions = conditions
+		opStatus.Phase = common.PhaseFailed
+		if err = operation.UpdateOperationApiStatus(object, opStatus); err != nil {
+			return utils.RequeueIfError(err)
+		}
+		return utils.RequeueImmediately()
+	}
+
+	// 1. update status to pending
+	opStatus.Phase = common.PhasePending
+	if len(opStatus.Conditions) == 0 {
+		opStatus.Conditions = []datav1alpha1.Condition{}
+	}
+	opStatus.Duration = "Unfinished"
+	opStatus.Infos = map[string]string{}
+	if err = operation.UpdateOperationApiStatus(object, opStatus); err != nil {
+		log.Error(err, fmt.Sprintf("failed to update the %s", operation.GetOperationType()))
+		return utils.RequeueIfError(err)
+	}
+	log.V(1).Info(fmt.Sprintf("Update phase of the %s to Pending successfully", operation.GetOperationType()))
+	return utils.RequeueImmediately()
+}
+
+func (t *TemplateEngine) reconcilePending(ctx cruntime.ReconcileRequestContext, object client.Object,
+	opStatus *datav1alpha1.OperationStatus, operation dataoperation.OperationInterface) (ctrl.Result, error) {
+	log := ctx.Log.WithName("reconcilePending")
+
+	// 1. lock the dataset
+	err := SetDataOperationInTargetDataset(ctx, object, operation, t)
+	if err != nil {
+		return utils.RequeueAfterInterval(20 * time.Second)
+	}
+
+	log.Info("Get lock on target dataset, try to update phase")
+	opStatus.Phase = common.PhaseExecuting
+	if err = operation.UpdateOperationApiStatus(object, opStatus); err != nil {
+		log.Error(err, fmt.Sprintf("failed to update %s status to Executing, will retry", operation.GetOperationType()))
+		return utils.RequeueIfError(err)
+	}
+	log.V(1).Info(fmt.Sprintf("update %s status to Executing successfully", operation.GetOperationType()))
+	return utils.RequeueImmediately()
+}
+
+func (t *TemplateEngine) reconcileExecuting(ctx cruntime.ReconcileRequestContext, object client.Object,
+	opStatus *datav1alpha1.OperationStatus, operation dataoperation.OperationInterface) (ctrl.Result, error) {
+	log := ctx.Log.WithName("reconcileExecuting")
+
+	// 1. Install the helm chart if not exists
+	support, err := InstallDataOperationHelmIfNotExist(ctx, object, operation, t.Implement)
```

Comment:
Why don't we make "not supported" a specific error here?

Reply:
fixed.